src/libraries/System.Private.CoreLib/src/System/AppContextConfigHelper.cs

```diff
@@ -15,7 +15,7 @@ internal static bool GetBooleanConfig(string switchName, string envVariable, boo
             if (!AppContext.TryGetSwitch(switchName, out bool ret))
             {
                 string? switchValue = Environment.GetEnvironmentVariable(envVariable);
-                ret = switchValue != null ? (bool.IsTrueStringIgnoreCase(switchValue) || switchValue.Equals("1")) : defaultValue;
+                ret = switchValue != null ? (switchValue.Equals("true", StringComparison.OrdinalIgnoreCase) || switchValue.Equals("1")) : defaultValue;
```

Comment:
It is dangerous to call `string.Equals(..., StringComparison.OrdinalIgnoreCase)` here. It can call into globalization stack when the JIT optimizations does not kick in, and the globalization will call this method recursively to read its config.

Reply:
ah, good point, I see that `Ordinal.EqualsIgnoreCase` has a fast path for ASCII but if the value in the config is not ASCII it might trigger globalization indeed, let me restore the previous version here

Reply:
Good catch, I'm sad I was not smart enough to see it myself 😞

Reply:
It would be also possible to fix this by changing OrdinalIgnoreCase to avoid initializing globalization.

OrdinalIgnoreCase will fallback to the slow path if one of the chars is not ASCII. It really only needs to fallback to the slow path if both chars are not ASCII.

Reply:
@jkotas do you mind if I investigate it separately? I also wanted to look into vectorizing `Ordinal.CompareStringIgnoreCase` - it currently uses only SWAR version

Reply:
Yes, that's fine

Reply:
Mono tests are still failing though. You may need to fix this to make them pass.